src/helics/shared_api_library/helicsCallbacks.h

```diff
@@ -140,6 +140,28 @@ HELICS_EXPORT void helicsFederateSetTimeUpdateCallback(HelicsFederate fed,
                                                        void* userdata,
                                                        HelicsError* err);
 
+/**
+ * Set callback for the federate mode change.
+ *
+ * @details This callback will be executed every time the operating mode of the federate changes.
+ *
+ * @param fed The federate to set the callback for.
+ * @param timeUpdate A callback with signature void(HelicsTime newTime, bool iterating, void *userdata);
+ *                 The function arguments are the new time value, a bool indicating that the time is iterating, and pointer to the userdata.
```

Comment:
I think this doc string for the C API needs updating to match the mode change function callback.